packages/firestore/test/unit/specs/bundle_spec.test.ts

```diff
@@ -0,0 +1,349 @@
+  );
+  if (testDoc.createTime) {
+    builder.addDocument(
+      testDoc.key,
+      toVersion(JSON_SERIALIZER, version(testDoc.createTime)),
+      toVersion(JSON_SERIALIZER, version(testDoc.updateTime!)),
+      testDoc.content!
+    );
+  }
+  return builder.build(
+    'test-bundle',
+    toVersion(JSON_SERIALIZER, version(testDoc.readTime))
+  );
+}
+
+describeSpec('Bundles:', ['no-ios', 'no-android'], () => {
+  specTest('Newer docs from bundles should overwrite cache', [], () => {
+    const query1 = Query.atPath(path('collection'));
+    const docA = doc('collection/a', 1000, { key: 'a' });
+    const docAChanged = doc('collection/a', 2999, { key: 'b' });
+
+    const bundleString = bundleWithDocument({
+      key: docA.key,
+      readTime: 3000,
+      createTime: 1999,
+      updateTime: 2999,
+      content: { key: { stringValue: 'b' } }
+    });
+
+    return spec()
+      .userListens(query1)
+      .watchAcksFull(query1, 1000, docA)
+      .expectEvents(query1, { added: [docA] })
+      .loadBundle(bundleString)
+      .expectEvents(query1, { modified: [docAChanged], fromCache: true });
+  });
+
+  specTest('Newer deleted docs from bundles should delete cache', [], () => {
+    const query1 = Query.atPath(path('collection'));
+    const docA = doc('collection/a', 1000, { key: 'a' });
+
+    const bundleString = bundleWithDocument({
+      key: docA.key,
+      readTime: 3000
+    });
+
+    return spec()
+      .userListens(query1)
+      .watchAcksFull(query1, 1000, docA)
+      .expectEvents(query1, { added: [docA] })
+      .loadBundle(bundleString)
+      .expectEvents(query1, { removed: [docA], fromCache: true });
+  });
+
+  specTest('Older deleted docs from bundles should do nothing', [], () => {
+    const query1 = Query.atPath(path('collection'));
+    const docA = doc('collection/a', 1000, { key: 'a' });
+
+    const bundleString = bundleWithDocument({
+      key: docA.key,
+      readTime: 999
+    });
+
+    return (
+      spec()
+        .userListens(query1)
+        .watchAcksFull(query1, 1000, docA)
+        .expectEvents(query1, { added: [docA] })
+        // No events are expected here.
+        .loadBundle(bundleString)
+    );
+  });
+
+  specTest(
+    'Newer docs from bundles should raise snapshot only when watch catches up with acknowledged writes',
+    [],
+    () => {
+      const query = Query.atPath(path('collection'));
+      const docA = doc('collection/a', 250, { key: 'a' });
+
+      const bundleBeforeMutationAck = bundleWithDocument({
+        key: docA.key,
+        readTime: 500,
+        createTime: 250,
+        updateTime: 500,
+        content: { key: { stringValue: 'b' } }
+      });
+
+      const bundleAfterMutationAck = bundleWithDocument({
+        key: docA.key,
+        readTime: 1001,
+        createTime: 250,
+        updateTime: 1001,
+        content: { key: { stringValue: 'fromBundle' } }
+      });
+      return (
+        spec()
+          .withGCEnabled(false)
```

Comment:
Can you file a bug to define the behavior bundles when `withGCEnabled` is not set to false? Note that this flag is spec test only and that this not actually a mode that users can enable with memory persistence.